pkg/cloud/services/networking/securitygroups.go

```diff
@@ -147,51 +153,57 @@ func generateControlPlaneGroup(clusterName string) infrav1.SecurityGroup {
 					RemoteIPPrefix: "0.0.0.0/0",
 				},
 				{
-					Direction:      "ingress",
-					EtherType:      "IPv4",
-					PortRangeMin:   22,
-					PortRangeMax:   22,
-					Protocol:       "tcp",
-					RemoteIPPrefix: "0.0.0.0/0",
+					Direction:     "ingress",
+					EtherType:     "IPv4",
+					PortRangeMin:  2379,
+					PortRangeMax:  2380,
+					Protocol:      "tcp",
+					RemoteGroupID: remoteGroupIDSelf,
+				},
+				{
+					Direction:     "ingress",
+					EtherType:     "IPv4",
+					PortRangeMin:  10250,
+					PortRangeMax:  10250,
+					Protocol:      "tcp",
+					RemoteGroupID: remoteGroupIDSelf,
 				},
 			},
 			defaultRules...,
 		),
 	}
 }
 
-func generateGlobalGroup(clusterName string) infrav1.SecurityGroup {
-	secGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, globalSuffix)
+func (s *Service) generateWorkerGroup(clusterName string) infrav1.SecurityGroup {
+	secWorkerGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, workerSuffix)
+	secControlPlaneGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, controlPlaneSuffix)
+
+	secControlPlaneGroup, err := s.getSecurityGroupByName(secControlPlaneGroupName)
+	if err != nil || secControlPlaneGroup == nil || secControlPlaneGroup.ID == "" {
+		return infrav1.SecurityGroup{}
+	}
 
 	// As above, hardcoded rules.
 
 	return infrav1.SecurityGroup{
-		Name: secGroupName,
+		Name: secWorkerGroupName,
 		Rules: append(
 			[]infrav1.SecurityGroupRule{
 				{
```

Comment:
Looks to me like the workers cannot communicate with each other anymore? If I'm right, isn't that a problem? 
Also how does the communication between control plane and workers and back works now?

Reply:
I understand I am missing considering the cni plugin. Let's add the rule to allow workers to communicate each other.
In terms of the communication between controller and worker, controller accept 6443/tcp(kube-apiserver) from all including worker nodes, and worker accepts 10250/tcp(kubelet API) from controller. This should work fine.

Reply:
we might have more workers after the init cluster deploy, will this also considered?
maybe need some detail thought on what we expect the cluster network to be provisioned by default and guide user to add advanced by their own..

Reply:
@hidekazuna worker<=> master sounds good. So you also add a rule that the worker nodes can communicate with each other, right?

Reply:
@sbueringer Yes, I will also add the rule you mentioned.

Reply:
@sbueringer Ther rule was added here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/555/files#diff-981e50f3b13da070546823dd24e18b32R197-R211

Reply:
@sbueringer @jichenjc I studied calico and skeptical about adding the rule for worker communication now. Calico default networking requirement is bidirectional tcp 179 and IPinIP. I created new security group including the rule allowing bidirectional tcp 179 and IPinIP, and added the group to both controllers and workers. It is working fine.
We should leave the issue due to depending on the CNI implementation.

Reply:
I am not fully understand the comments above
>>>I created new security group including the rule allowing bidirectional tcp 179 and IPinIP, and added the group to both controllers and workers. 

you mean after your change, the cluster still not work and you have to create the rules manually? as we are provide solution to `create a k8s cluster` so we should consider how to enhance it , if it's a generic requirements , we need talk to cluster-api folks?

Reply:
All nodes are ready status, but pod to pod communication is not good. I think it is not good before my change. Kubespray doc says  [in OpenStack you must allow ipip traffic in your security groups ](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/calico.md#cloud-providers-configuration).

Reply:
Calico is just an example for Cluster API. So we should leave the issue for users.

Reply:
@sbueringer @jichenjc I found [cluster-api-provider-aws implementation](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/services/ec2/securitygroups.go#L368-L488)! I will follow this implementaion 😃

Reply:
seems that's for calico only :)

Reply:
now we understand cluster api is calico only :)

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3174 opened

Reply:
Okay so if calico works now, it's fine for me to resolve this conversation.

Reply:
Calico is not working as of now. Pod to pod communication does not work. I will fix.